File: scripts/download-banner-video.js
import fs from 'fs';
import path from 'path';

const videoUrl = 'https://hebbkx1anhila5yf.public.blob.vercel-storage.com/2%E6%9C%8814%E6%97%A5%20%281%29-UNFfSjj1SKya5DDcw8S7Mo5wkWahjb.mp4';
const outputDir = '/vercel/share/v0-project/public/videos/banner';
const outputPath = path.join(outputDir, 'banner-1.mp4');

console.log('[v0] 开始下载banner视频...');
console.log('[v0] 输出路径:', outputPath);

// 下载视频
async function downloadVideo() {
  try {
    console.log('[v0] 正在从CDN下载视频...');
    const response = await fetch(videoUrl);
    
    if (!response.ok) {
      throw new Error(`HTTP ${response.status}: ${response.statusText}`);
    }

    const buffer = await response.arrayBuffer();
    const bytes = Buffer.from(buffer);
    
    // 使用流式写入
    const writeStream = fs.createWriteStream(outputPath);
    
    writeStream.write(bytes);
    writeStream.end();
    
    writeStream.on('finish', () => {
      console.log(`[v0] ✓ 视频下载完成！`);
      console.log(`[v0] 文件大小: ${(bytes.length / 1024 / 1024).toFixed(2)} MB`);
      console.log(`[v0] 保存位置: ${outputPath}`);
      process.exit(0);
    });
    
    writeStream.on('error', (err) => {
      console.error('[v0] ✗ 文件写入失败:', err.message);
      process.exit(1);
    });
  } catch (err) {
    console.error('[v0] ✗ 下载失败:', err.message);
    process.exit(1);
  }
}

// 执行下载
downloadVideo();

File: scripts/download_banner_video.py
#!/usr/bin/env python3
import urllib.request
import os
from pathlib import Path

# 视频URL
VIDEO_URL = "https://hebbkx1anhila5yf.public.blob.vercel-storage.com/2%E6%9C%8814%E6%97%A5%20%281%29-UNFfSjj1SKya5DDcw8S7Mo5wkWahjb.mp4"

# 本地保存路径
PROJECT_DIR = Path("/vercel/share/v0-project")
PUBLIC_DIR = PROJECT_DIR / "public" / "images" / "banner"
OUTPUT_FILE = PUBLIC_DIR / "banner-video.mp4"

# 创建目录（如果不存在）
PUBLIC_DIR.mkdir(parents=True, exist_ok=True)

print(f"[v0] 开始下载视频...")
print(f"[v0] 源URL: {VIDEO_URL}")
print(f"[v0] 保存路径: {OUTPUT_FILE}")

try:
    # 下载视频
    urllib.request.urlretrieve(VIDEO_URL, OUTPUT_FILE)
    
    # 验证文件大小
    file_size = os.path.getsize(OUTPUT_FILE)
    print(f"[v0] 视频下载成功！")
    print(f"[v0] 文件大小: {file_size / (1024*1024):.2f} MB")
    print(f"[v0] 文件路径: {OUTPUT_FILE}")
    
except Exception as e:
    print(f"[v0] 错误: 下载失败 - {str(e)}")
    exit(1)

print(f"[v0] 完成！")
